on:
  push:
    branches: [ '*' ]

jobs:
  test:

    runs-on: ubuntu-latest

    steps:
    - uses: actions/checkout@v2
      with:
        fetch-depth: 0
    - name: Set up Python 3.9
      uses: actions/setup-python@v1
      with:
        python-version: 3.9
    - name: Detect Challenge Directory
      run: |
        echo "Branch: $(git branch --show-current)"
        CHALLENGE_DIR=`git diff refs/remotes/origin/master HEAD --name-only | python determine_challenge_directory.py`
        echo "CHALLENGE_DIR: $CHALLENGE_DIR"
        echo "CHALLENGE_DIR=$CHALLENGE_DIR" >> $GITHUB_ENV
    - name: Cache pip
      uses: actions/cache@v1
      with:
        path: ~/.cache/pip
        # Look to see if there is a cache hit for the corresponding requirements file
        key: ${{ runner.os }}-pip-${{ hashFiles('requirements.txt') }}
        restore-keys: |
          ${{ runner.os }}-pip-
          ${{ runner.os }}-
    - name: Install dependencies
      run: |
        # libldap2-dev and libsasl2-dev are for ldap
        sudo apt-get update
        sudo apt-get install libldap2-dev libsasl2-dev
        pip install -r $CHALLENGE_DIR/requirements.dev.txt
    - name: Lint with flake8
      run: |
        cd $CHALLENGE_DIR/src/backend && flake8
    - name: Test with pytest
      run: |
        # make some dummy files so tests run fine
        sudo mkdir -p frontend/generated/static
        cd $CHALLENGE_DIR/src/backend
        ./manage.py collectstatic --noinput
        pytest --cov-fail-under=100
